src/control/lib/control/system.go

```diff
@@ -639,7 +641,24 @@ func LeaderQuery(ctx context.Context, rpcClient UnaryInvoker, req *LeaderQueryRe
 	}
 
 	resp := new(LeaderQueryResp)
-	return resp, convertMSResponse(ur, resp)
+	if err = convertMSResponse(ur, resp); err != nil {
+		return nil, errors.Wrap(err, "converting MS to LeaderQuery resp")
+	}
+
+	req.SetHostList(resp.Replicas)
+	ur, err = rpcClient.InvokeUnaryRPC(ctx, req)
```

Comment:
One thing I find myself wondering is whether the MS could find this information on the first call, rather than requiring two RPCs. Shouldn't the MS have that information in its database?

Reply:
Yes, it's store in the database. But we currently store the only host name and not the status of the raft. So if replicas are down, we need to send another RPC to those replicas to make sure all are up and responsive.

Reply:
Thanks for the explanation.